# Q1 - Find the mean of old scheme and new scheme column.

In [1]:
import pandas as pd
import numpy as np
from statsmodels.stats.power import ttest_power

In [2]:
df = pd.read_csv('Titan_insur.csv')

In [3]:
df.head()

SALESPERSON  Old_Scheme  New_Scheme
0            1          57          62
1            2         103         122
2            3          59          54
3            4          75          82
4            5          84          84

In [19]:
df.Old_Scheme.mean()

68.03333333333333

In [20]:
df.New_Scheme.mean()

72.03333333333333

Mean of old scheme - 68.03
Mean of New Scheme - 72.03

# Q2 - Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs?

In [6]:
from scipy.stats import ttest_1samp, ttest_ind,levene, shapiro

In [7]:
pre = df['Old_Scheme']

In [8]:
post =df['New_Scheme']

In [9]:
shapiro(pre)  # Null Hypothesis : Data is normal , Alternate H - Data is not normal, P(0.98) > 0.05 means data is normal

(0.9885101914405823, 0.9813658595085144)

In [10]:
shapiro(post) # Null Hypothesis : Data is normal , Alternate H - Data is not normal, P(0.50) > 0.05 means data is normal

(0.9687567353248596, 0.5057420134544373)

In [11]:
#Null Hypothesis - varaince of both groups are equal
#Alternate Hypothese - Variances are not equal
levene(pre,post)

LeveneResult(statistic=1.063061539437244, pvalue=0.30679836081811235)

In [12]:
# Inference - The data is Normal , paired and parametric Two sample T Test
# We will perform ttest_1samp
# Ho - Titan takes up the new scheme
# Ha - Titan do not take the new scheme

t_statistic, p_value = ttest_1samp(post - pre, 0) 
print(t_statistic, p_value)
print ("paired t-test p-value=", p_value)

1.5559143823544377 0.13057553961337662
paired t-test p-value= 0.13057553961337662


# Q3 - What conclusion does the test (p-value) lead to?

From the P value we can conclude that we fail to reject the Null Hypothesis that is Titan can continue with its new scheme 
and if would lead to profits.
As P value is higher than rejection percentage (5%)

# Q-4 Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme.If this figure is alternative hypothesis, what is:

a) The probability of a type 1 error? 

The probability of making a type I error is α, which is the level of significance you set for your hypothesis test. 
An α of 0.05 indicates that you are willing to accept a 5% chance that you are wrong when you reject the null hypothesis.

b)What is the p- value of the hypothesis test if we test for a difference of $5000?

In [18]:
# As we have assumed old schem , new scheme in thousands. Taking difference of $5000 as 5

t_statistic, p_value = ttest_1samp(post - pre, 5) 
print(t_statistic, p_value)
print ("paired t-test p-value=", p_value)

-0.3889785955886094 0.7001334912613286
paired t-test p-value= 0.7001334912613286


c) Power of the test

In [17]:
(np.mean(post) - np.mean(pre)) / np.sqrt(((30-1)*np.var(pre)+(30-1)*np.var(post)) / 30+30-2)

0.12909555513169643

In [15]:
print(ttest_power(0.129, nobs=30, alpha=0.05, alternative='larger'))

0.1698663025719903


Here the power of test value is quite low as we failed to reject the null hypothesis.